tests/swarm-deploy/test_swarm_runs.py

```diff
@@ -118,37 +116,73 @@ def test_core_service_running(
     # ID                  NAME                     IMAGE                     NODE                DESIRED STATE       CURRENT STATE            ERROR                       PORTS
     # puiaevvmtbs1        simcore_storage.1       simcore_storage:latest   crespo-wkstn        Running             Running 18 minutes ago
     # j5xtlrnn684y         \_ simcore_storage.1   simcore_storage:latest   crespo-wkstn        Shutdown            Failed 18 minutes ago    "task: non-zero exit (1)"
-    tasks = service.tasks()
-    num_tasks = get_replicas(service_config)
-
-    assert len(tasks) == num_tasks, (
-        f"Expected a {num_tasks} task(s) for '{service_name}', got instead"
-        f"\n{ get_tasks_summary(tasks) }"
-        f"\n{ get_failed_tasks_logs(service, docker_client) }"
-    )
 
-    for i in range(num_tasks):
-        task: Dict[str, Any] = service.tasks()[i]
+    expected_replicas = get_replicas(service_config)
 
-        for n in range(RETRY_COUNT):
-            task = service.tasks()[i]
+    # assert number of slots equals number of expected services
+    tasks = service.tasks()
+    slots: Set[int] = {t["Slot"] for t in tasks}
+    slot_count = len(slots)
+
+    assert slot_count == expected_replicas, (
+        f"Expected to have {expected_replicas} slot(s), "
+        f"instead {slot_count} slot(s) were found."

```

Comment:
same here. Please check this [article](https://docs.docker.com/engine/swarm/how-swarm-mode-works/services/) it says a slot is "analogous" to a task. and it is nowhere visible when issuing ```docker service ps```.